exponentation_by_squaring/exponentation_by_squaring.c

```diff
@@ -0,0 +1,25 @@
+#include<stdio.h>
+
+/* Algorithm to calculate x raised to power k very efficiently
+*  Can be used as a great alternative to normal pow function
+*  base - base of the expression
+*  power - power of expression
+*/
+
+long int exponentation_by_squaring(int base, int power) {
```

Comment:
WIll this definition be good?
```C
long double exponentation_by_squaring(long double, int);
```
In this way, we can easily pass integer and float type number as first parameter and get result as `long double`, which can be casted to desired type.